Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -136,6 +140,19 @@ eventPayloadHandler merchantOpCityId DST.OnRideCompletion {..} = do
           pure (fromMaybe 0.0 ride.fare, incrementBonusEarningsBy, incrementLateNightTripsCountBy, 10)
     -- Esq.runNoTransaction $ do
     DSQ.incrementTotalEarningsAndBonusEarnedAndLateNightTrip (cast driverId) incrementTotalEarningsBy (incrementBonusEarningsBy + overallPickupCharges) incrementLateNightTripsCountBy
+
+    -- for extra fare mitigation --
+    merchantConfig <- SCTC.findByMerchantOpCityId merchantOpCityId (Just (DriverId (cast driverId))) >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
+    booking <- B.runInReplica $ BQ.findById ride.bookingId >>= fromMaybeM (BookingNotFound ride.bookingId.getId)
```

Comment:
Booking Query also better if it can be avoided.

Reply:
Need to get booking stier to make it configurable against tier